kubernetes/go-gitops/waypoint.hcl

```diff
@@ -0,0 +1,63 @@
+project = "go-gitops-0"
+
+variable "image" {
+  default     = "your.own.container.registry.example.com/go"
+  type        = string
+  description = "Image name for the built image in the Docker registry."
+}
+
+variable "tag" {
+  default     = "latest"
+  type        = string
+  description = "Image tag for the image"
+}
+
+// set your registry username through the UI or on the CLI!
+variable "registry_username" {
+  default     = ""
+  type        = string
+  description = "username for container registry"
+}
+
+// set your registry password through the UI or on the CLI!
+variable "registry_password" {
+  default     = ""
+  type        = string
+  description = "password for registry" // DO NOT COMMIT YOUR PASSWORD TO GIT
+}
+
+variable "regcred_secret" {
+  default     = "regcred"
+  type        = string
+  description = "The existing secret name inside Kubernetes for authenticating to the container registry"
+}
```

Comment:
How is this established in the Kube cluster? Should we document it in the README?

Reply:
Before I set up the secret, I authenticate with docker to my container registry. Then I do something like this, but it's a generic kubernetics secret named `regcred`

```
kubectl create secret generic regcred --from-file=.dockerconfigjson=$HOME/.docker/config.json --type=kubernetes.io/dockerconfigjson
```

Not required I don't think for every workflow, just depends on how they're using container registries!